### PR TITLE
Docusaurus versioning 

### PR DESCRIPTION
**Pull request by samricotta**
Within in the sdk repo, versioning is split between vuepress and docusaurus. This PR migrates all versions over to this repo whilst using the custom plugin to build the docs per version. 

This PR is to be merged before https://github.com/cosmos/cosmos-sdk-docs/pull/1.

There will be another PR that will live in the sdk repo to copy over any changes on MD files using a GitHub Actions workflow which is triggered on any deployment.

### CODE REVIEW

**Comment by julienrbrt** [comment]
This PR is based on #1 but the description says it should be merged before #1, is that a typo?

**Comment by samricotta** [comment]
> This PR is based on #1 but the description says it should be merged before #1, is that a typo?

I think it will be best to follow your suggestion and configure without domain so I dont think my comment here matters. It was only that I wanted everything set up before going live.

File: custom-version-plugin.js
[the remark this answers is not in the document]
Comment (samricotta) [reply custom-version-plugin.js:1]: This is just the action they use in docusaurus, i was having trouble last night getting it to build so trying all options. The custom plugin is specified in `docusaurus.config.js`. It uses all the files from main and any of those that arent the same, they have their own version. Additionally, there is version_config which removes specific files or paths from versions. This way its very clear as to what is in each version when its built

**Comment by samricotta** [comment]
@julienrbrt Yeah this is why it needs to go on staging so i can literally proof read every single page and compare as there are still a lot of manual changes to happen

File: custom-version-plugin.js
[the remark this answers is not in the document]
Comment (julienrbrt) [reply custom-version-plugin.js:1]: I am not seeing it in the `docusaurus.config.js`. Which `main` are we talking about? This repo?
> It uses all the files from main and any of those that arent the same, they have their own version.
Ok, so the process of adding a custom content for a version is manual right? If so, this means the versioning is definitely incomplete, won't it be a hell to maintain?
Again, only if my understanding is correct:  If I want to change the content only for main, does not this mean I need to first copy the old content to each of the versioned SDK versions, where I want the content to stay the same and then edit the latest content?

**Comment by tac0turtle** [comment]
Hey checking in on this. Are we good to merge?

**Pull request merged**